unit-tests/src/org/commcare/CommCareTestApplication.java

```diff
@@ -214,4 +222,25 @@ public ModernHttpRequester buildModernHttpRequester(Context context, URL url,
     public DataPullRequester getDataPullRequester() {
         return LocalDataPullResponseFactory.INSTANCE;
     }
+
+    @Override
+    public void afterTest(Method method) {
+        Robolectric.flushBackgroundThreadScheduler();
+        if (asyncExceptions.size() > 0) {
+            for(Throwable throwable: asyncExceptions) {
+                throwable.printStackTrace();
+                fail("Test failed with " + asyncExceptions.size() + " asyncronous exceptions.");
```

Comment:
I think we actually want to display the exception itself as the failure. Isn't there a "Fail" message which directly takes in an exception?